CRM/Core/ManagedEntities.php

```diff
@@ -237,7 +237,7 @@ public function reconcileUnknownModules() {
    */
   public function insertNewEntity($todo) {
     $result = civicrm_api($todo['entity'], 'create', $todo['params']);
-    if ($result['is_error']) {
+    if (!empty($result['is_error'])) {
       $this->onApiError($todo['entity'], 'create', $todo['params'], $result);
```

Comment:
Does my read of this match what you're saying:
When api4 succeeds, there is no expected is_error element (because it's an object instead).
When api4 fails, there _should_ still be an array with $result['is_error']=1. (That's not quite what I get when I try it, but maybe that's the gap you mean.)

Reply:
I think api4 throws an exception & would bypass this anyway.

Reply:
api4 by itself might throw exception but this is the wrapper civicrm_api() which doesn't. I think this is ok just when I tested I got some other kind of error - sorry that's not the most detailed bug report but I didn't want to spend a lot of time on it if it was already a known "gap" since you mentioned there were some.